src/Task.cc

```diff
@@ -487,6 +487,33 @@ void Task::on_syscall_exit_arch(int syscallno, const Registers& regs) {
           }
           break;
         }
+        case PTRACE_ARCH_PRCTL: {
+          int code = (int)regs.arg4();
+          switch (code) {
+            case ARCH_GET_FS:
+            case ARCH_GET_GS:
+              break;
+            case ARCH_SET_FS:
+            case ARCH_SET_GS: {
+              Registers r = tracee->regs();
+              if (regs.arg3() == 0) {
+                // Work around a kernel bug in pre-3.7 kernels, where setting
```

Comment:
Hah, yeah. Have made that mistake more than once. Fixed locally, will push together with anything else once you're done reviewing.